src/core/expr/head_func_cut.cc

```diff
@@ -0,0 +1,221 @@
+                              "got `nbins[" << i << "`]: `" << nbin << "`";
+      }
+
+      nbins[i++] = nbin;
+    }
+    xassert(i == ncols);
+
+  } else {
+    if (defined_nbins) {
+      nbins_default = py_nbins_.to_int32_strict();
+      if (nbins_default <= 0) {
+        throw ValueError() << "Number of bins must be positive, "
+                              "instead got: `" << nbins_default << "`";
+      }
+    }
+
+    for (size_t i = 0; i < ncols; ++i) {
+      nbins[i] = nbins_default;
+    }
+  }
+
+  // Cut workframe in-place
+  for (size_t i = 0; i < ncols; ++i) {
+    Column coli = wf.retrieve_column(i);
+    coli = Column(Cut_ColumnImpl::make(
+             std::move(coli), i, nbins[i], right_closed_
+           ));
+    wf.replace_column(i, std::move(coli));
+  }
+
+  return wf;
+}
+
+}}  // dt::expr
+
+
+namespace py {
+
+static oobj make_pyexpr(dt::expr::Op opcode, otuple targs, otuple tparams) {
+  size_t op = static_cast<size_t>(opcode);
+  return robj(Expr_Type).call({ oint(op), targs, tparams });
+}
+
+
+static oobj cut_frame(oobj arg0, oobj arg1, oobj arg2) {
+  auto slice_all = oslice(oslice::NA, oslice::NA, oslice::NA);
+  auto f_all = make_pyexpr(dt::expr::Op::COL,
+                           otuple{ slice_all },
+                           otuple{ oint(0) });
+  auto cutexpr = make_pyexpr(dt::expr::Op::CUT,
+                             otuple{ f_all },
+                             otuple{ arg1, arg2 });
+  auto frame = static_cast<Frame*>(arg0.to_borrowed_ref());
+  return frame->m__getitem__(otuple{ slice_all, cutexpr });
+}
+
+
+static const char* doc_cut =
+R"(cut(cols, nbins=10, right_closed=True)
+--
+
+Cut all the columns in a Frame/f-expression by binning
+their values into equal-width discrete intervals.
+
+Parameters
+----------
+cols: Frame | f-expression
+    Frame or f-expression consisting of numeric columns.
+nbins: int | list of ints | tuple of ints
+    When a single number is specified, this number of bins
+    will be used to bin each column of `cols`.
+    When a list or a tuple is provided, each column will be binned
+    by using its own number of bins. In the latter case,
+    the list/tuple length must be equal to the number of columns
+    in `cols`.
+right_closed: bool
+    Each binning interval is `half-open`_. This flag indicates which
+    side of the interval is closed.
+
+return: Frame | Expr
+    The return type matches the type of the `cols` argument.
+    If the function is applied to a frame, then the result is a frame where
+    each column from the original frame has been cut into the specified bins.
+    If the `cols` argument is an f-expression, then the result is a new
+    f-expression that transforms every column into its cut version.
+
+See also
+--------
+:func:`qcut()` -- function for quantile binning.
+
+.. _`half-open`: https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology
+
+)";
+
+
+static PKArgs args_cut(
+  1, 2, 0, false, false,
```

Comment:
Maybe this should be `1, 0, 2, ...` (or at least `1, 1, 1`): require arguments `nbins` and `right_closed` to be explicitly named. Consider call `cut(f.A, 10)` -- this seems ok, cut into 10 bins. But `cut(f.A, 10, True)` is already very confusing. Plus, once we have the `bins` argument as well, then it would be unclear why we require `bins=` to be explicitly named while allowing `nbins=` to stay unnamed.

Reply:
Sure, let's do it `1, 0, 2`.